### ISSUE TITLE
use-flow-type should apply when defining a type and not using it in the same file

### ISSUE THREAD

**Issue by Leeds-eBooks** [gajus/eslint-plugin-flowtype#136]
For convenience of locating my types, I put all my type aliases in separate files, and reference these files under the `[libs]` section of my `.flowconfig`. So I might have two files as follows:

```js
type T = {x: number}
```

```js
const obj: T = {x: 5}
```

`use-flow-type` doesn't handle this. It only suppresses eslint errors if the type and its usage-in-code are in the same file, **or** if the unused type is defined with the `declare` keyword.

The use of `declare` is not well documented – it is not required, even for declaring types in flow's "global" scope (i.e. so they are available throughout the project), as can be seen from [this Twitter conversation](https://twitter.com/lbljeffmo/status/788504721439031296), and from the flow docs [here](https://flowtype.org/docs/declarations.html#declaring-types) where it says that "Declared types include type aliases and interfaces; in fact, they share the same syntax as usual types, so the `declare` keyword is superfluous."

Simply put: `use-flow-type` should suppress warnings for type definitions even if they are not used in the current file, and even if they are not `declare`d.

**Comment by danharper** [comment]
I don't think that would be best. I'd want to _discourage_ global type definitions, and explicitly import/export only those I need to share.

Instead of putting your type files under `[libs]` in your `.flowconfig`, do this:

```js
// my-types.js

export type Foo = {}
```

```js
// my-source.js

import type {Foo} from './my-types'

const obj: Foo = {}
```

If you still want to stick to global definitions - use `declare type` instead to suppress them.

**Comment by Leeds-eBooks** [comment]
Fair enough, good point!
